Source/scrollrt.cpp

```diff
@@ -377,13 +356,74 @@ void DrawMonster(const Surface &out, int x, int y, int mx, int my, const Monster
 		return;
 	}
 
+	constexpr auto getMonsterModeDisplayName = [](MonsterMode monsterMode) {
+		switch (monsterMode) {
+		case MonsterMode::Stand:
+			return "standing";
+
+		case MonsterMode::MoveNorthwards:
+			return "moving (northwards)";
+
+		case MonsterMode::MoveSouthwards:
+			return "moving (southwards)";
+
+		case MonsterMode::MoveSideways:
+			return "moving (sideways)";
+
+		case MonsterMode::MeleeAttack:
+			return "attacking (melee)";
+
+		case MonsterMode::HitRecovery:
+			return "getting hit";
+
+		case MonsterMode::Death:
+			return "dying";
+
+		case MonsterMode::SpecialMeleeAttack:
+			return "attacking (special melee)";
+
+		case MonsterMode::FadeIn:
+			return "fading in";
+
+		case MonsterMode::FadeOut:
+			return "fading out";
+
+		case MonsterMode::RangedAttack:
+			return "attacking (ranged)";
+
+		case MonsterMode::SpecialStand:
+			return "standing (special)";
+
+		case MonsterMode::SpecialRangedAttack:
+			return "attacking (special ranged)";
+
+		case MonsterMode::Delay:
+			return "delaying";
+
+		case MonsterMode::Charge:
+			return "charging";
+
+		case MonsterMode::Petrified:
+			return "petrified";
+
+		case MonsterMode::Heal:
+			return "healing";
+
+		case MonsterMode::Talk:
+			return "talking";
+
+		default:
+			app_fatal("Invalid monster mode.");

```

Comment:
This is only used for the log message, might as well return "unknown action" instead of triggering a fatal error,

Reply:
It should not be possible to ever hit this since there is a check right before it. I believe stopping is the best course of action here.

Reply:
nice work here :)

I would also have gone with `return "unknown action"` but it's an error case on top of an error case and I have never seen the first one so I don't think halting here is an issue.